docs/topic-guides/system-concepts/scheduling.txt

```diff
@@ -14,6 +24,21 @@ As new experiments become active or the resource demand of an active
 experiment changes, the scheduler will adjust how slots are allocated to
 experiments as appropriate.
 
-Scheduling behavior can be configured via the ``resources`` section of
-the experiment config file; see :ref:`experiment-configuration` for
-details.
+**********
+ Priority
+**********
+
+The master allocates cluster resources (slots) to active tasks based on
+their priority. While tasks of higher priority (lower priority number)
+are pending, no lower priority tasks will be scheduled. For instance, if
+tasks with priorities of five and forty-two are pending, the latter will
+not be scheduled until the prior has been. By default the priority
+scheduler will not preempt lower priority tasks in favor of higher
+priority tasks. If preemption is enabled, in scenarious where a higher
+priority task is pending and can not be scheduled, the scheduler will
+attempt to schedule it by pre-empting lower priority tasks, starting
+with the lowest priorities.
+
```

Comment:
This section could use an example or two. Maybe one kitchen sink sequence on an 8 GPU cluster, with preemption enabled:

T0: user submits priority 2 `adaptive_asha` experiment with `max_trials` 20 and `slots_per_trial` 1. 8 trials run and utilize all 8 GPUs
T1: user submits priority 1 dtrain experiment with slots_per_trial 4. 4 asha trials preempted so it can run. Note that if preemption were disabled, this would not get scheduled until the ASHA experiment's GPU demand becomes <= 4
T2: user starts priority 3 notebook with `resources.slots` 1. will run as soon as `adaptive_asha` plus dtrain experiments collectively need <= 7 GPUs
T3: asha and dtrain both complete
T4: user submits priority 1 dtrain experiment with `slots_per_trial` 8. will not be scheduled bc notebook not preemptible and only 7 slots available
T5: user submits priority 2 dtrain experiment with `slots_per_trial` 4. will still not be scheduled even though 7 slots available, because it is behind a higher priority task
T6: notebook killed. P1 dtrain runs.  Once complete, P2 dtrain runs

Reply:
This is a great example @davetroiano thanks for putting it together.